src/gui/plugins/align_tool/AlignTool.cc

```diff
@@ -0,0 +1,647 @@
+    /// or last entity selected.
+    public: bool first{true};
+
+    /// \brief The current selected entities
+    public: std::vector<Entity> selectedEntities;
+
+    /// \brief The previous positions of all the selected nodes.  Should always
+    /// be equal to the number of selected entities.
+    public: std::vector<math::Vector3d> prevPositions;
+
+    /// \brief The current queue of states to be executed.
+    public: std::queue<AlignState> states;
+
+    /// \brief The map of the original transparency values for the nodes.
+    public: std::map<std::string, double> originalTransparency;
+  };
+}
+
+using namespace ignition;
+using namespace gazebo;
+
+/////////////////////////////////////////////////
+AlignTool::AlignTool()
+  : GuiSystem(), dataPtr(std::make_unique<AlignToolPrivate>())
+{
+  // Deselect all entities upon loading plugin
+  auto deselectEvent = new gui::events::DeselectAllEntities(true);
+  ignition::gui::App()->sendEvent(
+      ignition::gui::App()->findChild<ignition::gui::MainWindow *>(),
+      deselectEvent);
+}
+
+/////////////////////////////////////////////////
+AlignTool::~AlignTool() = default;
+
+/////////////////////////////////////////////////
+void AlignTool::LoadConfig(const tinyxml2::XMLElement *)
+{
+  if (this->title.empty())
+    this->title = "Align tool";
+
+  // For align tool requests
+  ignition::gui::App()->findChild
+      <ignition::gui::MainWindow *>()->installEventFilter(this);
+}
+
+/////////////////////////////////////////////////
+void AlignTool::Update(const UpdateInfo &/* _info */,
+    EntityComponentManager &_ecm)
+{
+  std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
+  if (this->dataPtr->worldName.empty())
+  {
+    // TODO(anyone) Only one scene is supported for now
+    _ecm.Each<components::World, components::Name>(
+        [&](const Entity &/*_entity*/,
+          const components::World * /* _world */ ,
+          const components::Name *_name)->bool
+        {
+          this->dataPtr->worldName = _name->Data();
+          return true;
+        });
+  }
+}
+
+/////////////////////////////////////////////////
+void AlignTool::OnAlignConfig(const QString &_config)
+{
+  std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
+  std::string newConfig = _config.toStdString();
+  std::transform(newConfig.begin(), newConfig.end(),
+                 newConfig.begin(), ::tolower);
+
+  if (newConfig == "min")
+  {
+    this->dataPtr->config = AlignConfig::ALIGN_MIN;
+  }
+  else if (newConfig == "mid")
+  {
+    this->dataPtr->config = AlignConfig::ALIGN_MID;
+  }
+  else if (newConfig == "max")
+  {
+    this->dataPtr->config = AlignConfig::ALIGN_MAX;
+  }
+  else
+  {
+    ignwarn << "Invalid align axis config: " << newConfig << "\n";
+    ignwarn << "The valid options are:\n";
+    ignwarn << " - min\n";
+    ignwarn << " - mid\n";
+    ignwarn << " - max\n";
+  }
+}
+
+void AlignTool::OnReverse(bool _reverse)
```

Comment:
added `/////////////////////////////////////////////////`

Reply:
Resolved in [afe5d75b](https://github.com/ignitionrobotics/ign-gazebo/pull/59/commits/afe5d75b8f2ee1b8b38247903c54b5d596f02760)